vault/logical_system.go

```diff
@@ -301,7 +311,13 @@ func (b *SystemBackend) handlePluginCatalogRead(ctx context.Context, req *logica
 	if pluginName == "" {
 		return logical.ErrorResponse("missing plugin name"), nil
 	}
-	plugin, err := b.Core.pluginCatalog.Get(ctx, pluginName)
+
+	pluginType, err := consts.ParsePluginType(d.Get("type").(string))
+	if err != nil {
+		return nil, err
+	}
+
+	plugin, err := b.Core.pluginCatalog.Get(ctx, pluginName, pluginType)
```

Comment:
This comment doesn't fit well anywhere, but on line 373 below I need to go through the plugin reload code.